File: frontend/src/components/map/mapUtils.js
// src/components/map/mapUtils.js
import { mapa } from './mapData';

export function isCellWalkable(x, y) {
    if (y < 0 || y >= mapa.length || x < 0 || x >= mapa[0].length) return false;
    return mapa[y][x] !== 'X';  // 'X' é obstáculo
}

export function getCellType(x, y) {
    if (y < 0 || y >= mapa.length || x < 0 || x >= mapa[0].length) return null;
    return mapa[y][x];
}

File: backend/core/mapa/map_utils.py
from core.mapa.map_data import mapa

def is_cell_walkable(x, y):
    return 0 <= y < len(mapa) and 0 <= x < len(mapa[0]) and mapa[y][x] != 'X'

def get_cell_type(x, y):
    if 0 <= y < len(mapa) and 0 <= x < len(mapa[0]):
        return mapa[y][x]
    return None